testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -443,6 +443,21 @@ def test_gaps(self, universe, Writer, outfile):
                 with pytest.raises(mda.NoDataError):
                     getattr(written_ts, 'velocities')
 
+        
+    def test_data(self, universe, Writer, outfile):
+        with Writer(outfile, universe.atoms.n_atoms, dt=universe.trajectory.dt) as W:
+            for ts in universe.trajectory:
+                W.write(universe)
+
+        uw = mda.Universe(GRO, outfile)
+
+        # check that the data are identical for each time step
+        for orig_ts, written_ts in zip(universe.trajectory, uw.trajectory):
+            assert orig_ts.data == written_ts.data, ('mismatch between '
```

Comment:
This will also pass if both are 'None' or empty. Can we also check that 'ts.data' contains the data we expect? 

Follow up question is there data in ts.data for the original file?

Reply:
`ts.data` contains this information. I was just patching the behavior described in #3350 with the solution described by https://github.com/MDAnalysis/mdanalysis/issues/3350#issuecomment-864844782 which indicates that `ts.data['step']` might not exist in some cases. I can check the contents of the test file, if that's what you mean.

```
{'time': 0.0, 'step': 0, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 100.00000762939453, 'step': 50000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 200.00001525878906, 'step': 100000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 300.0, 'step': 150000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 400.0000305175781, 'step': 200000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 500.0000305175781, 'step': 250000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 600.0, 'step': 300000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 700.0000610351562, 'step': 350000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 800.0000610351562, 'step': 400000, 'lambda': 0.0, 'dt': 100.00000762939453}
{'time': 900.0000610351562, 'step': 450000, 'lambda': 0.0, 'dt': 100.00000762939453}
```

Reply:
Sorry I should have been clearer, my bad, was just checking that `ts.data` has something in it in the test file. All sorted.